tools/ontology/codegen/src/main/java/org/visallo/tools/ontology/ingest/codegen/CodegenCLI.java

```
@@ -0,0 +1,300 @@
+                writer.println("  public " + propertyAdditionType +
+                        " add" + upperCamelCasePropertyName + "(String key, Object value)" +
+                        " { return " + helperMethodName + "(" + constantName + ".getPropertyName(), key, value); }");
+            }
+
+            LOGGER.debug("  %s property %s", propertyType, upperCamelCasePropertyName);
+        });
+    }
+
+    private List<ClientApiOntology.Property> findPropertiesByIri(Map<String, ClientApiOntology.Property> propertyMap, List<String> propertyIris) {
+        return propertyIris.stream().sorted()
+                .map(propertyIri -> {
+                    ClientApiOntology.Property property = propertyMap.get(propertyIri);
+                    if (property == null) {
+                        throw new VisalloException("Unable to locate property for iri: " + propertyIri);
+                    }
+                    return property;
+                })
+                .collect(Collectors.toList());
+    }
+
+    private String constantNameFromClassName(String className) {
+        String[] classNameParts = StringUtils.splitByCharacterTypeCamelCase(className);
+        return Arrays.stream(classNameParts)
+                .map(String::toUpperCase)
+                .collect(Collectors.joining("_"));
+    }
+
+
+    private String classNameFromIri(String iri) {
+        Matcher matcher = IRI_FORMAT.matcher(iri);
+        if (matcher.matches()) {
+            String[] classNameParts = StringUtils.splitByCharacterTypeCamelCase(matcher.group(2));
+            return Arrays.stream(classNameParts)
+                    .filter(classNamePart -> CLASSNAME_PART_MATCHER.matcher(classNamePart).matches())
+                    .map(StringUtils::capitalize)
+                    .collect(Collectors.joining(""));
+        } else {
+            LOGGER.error("Unsupported iri pattern %s", iri);
```

Do we want to throw an exception here?